File: VB/WindowsApplication1/MyRibbonControl/MyRibbonControl.vb
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.ViewInfo

Namespace WindowsApplication1
	Public Class MyRibbonControl
		Inherits RibbonControl
		Public Sub New()
		End Sub

		Protected Overrides Function CreateViewInfo() As RibbonViewInfo
			Return New MyRibbonViewInfo(Me)
		End Function

		Public Delegate Sub CanMinimizePageGroupEventHandler(ByVal sender As Object, ByVal e As CanMinimizePageGroupEventArgs)
		Public Event CanMinimizePageGroup As CanMinimizePageGroupEventHandler

		Friend Function RaiseCanMinimizePageGroup(ByVal group As RibbonPageGroup, ByVal panelViewInfo As RibbonPanelViewInfo) As Boolean
			Dim args As New CanMinimizePageGroupEventArgs(True, group, panelViewInfo)
			RaiseEvent CanMinimizePageGroup(Me, args)

			Return args.CanMinimize
		End Function

	End Class
End Namespace

File: CS/WindowsApplication1/MyRibbonControl/MyRibbonControl.cs
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.ViewInfo;

namespace WindowsApplication1
{
    public class MyRibbonControl : RibbonControl
    {
        public MyRibbonControl()
        {
        }

        protected override RibbonViewInfo CreateViewInfo()
        {
            return new MyRibbonViewInfo(this);
        }

        public delegate void CanMinimizePageGroupEventHandler(object sender, CanMinimizePageGroupEventArgs e);
        public event CanMinimizePageGroupEventHandler CanMinimizePageGroup;

        internal bool RaiseCanMinimizePageGroup(RibbonPageGroup group, RibbonPanelViewInfo panelViewInfo)
        {
            CanMinimizePageGroupEventArgs args = new CanMinimizePageGroupEventArgs(true, group, panelViewInfo);
            if ( CanMinimizePageGroup != null )
                CanMinimizePageGroup(this, args);

            return args.CanMinimize;
        }

    }
}